keras_nlp/samplers/beam_sampler_test.py

```diff
@@ -42,11 +42,11 @@ def join_as_string(self, x):
         return ["".join([self.int_lookup[i] for i in s]) for s in x.numpy()]
 
     def test_stateless_call(self):
-        def next(prompt, state, index):
+        def next(prompt, cache, index):
             # Return a distribution favoring the first token in the vocab.
             logits = np.zeros((self.batch_size, self.vocab_size))
```

Comment:
```
# Return a distribution favoring the next char in cache.
hidden_state = np.zeros((self.batch_size, 1))
logits = tf.one_hot(hidden_state self.vocab_size) * 1e9
logits[:, 0] = 1e9
```

Reply:
Refactored the way to return one-hot vector, but i would prefer to keep logits and `hidden_state` decoupled, wdyt?

Reply:
sure!